usr/local/share/bastille/rename.sh

```diff
@@ -81,24 +81,7 @@ update_fstab() {
     # Update fstab to use the new name
     FSTAB_CONFIG="${bastille_jailsdir}/${NEWNAME}/fstab"
     if [ -f "${FSTAB_CONFIG}" ]; then
-        # Skip if fstab is empty, e.g newly created thick or clone jails
```

Comment:
Why are you removing all these lines?

Reply:
I don't have much familiarity with this code, I just fixed it to fit my use case, so let's go through it.

My proposal is to change occurrences of `${bastille_jailsdir}/${TARGET}` into `${bastille_jailsdir}/${NEWNAME}` in the fstab file. I think this is hardly debatable: we are renaming the jail `${TARGET}` into the jail `${NEWNAME}`, so all mount points that were pointing to the former now need to point to the latter.

So why did I remove all other lines? I think the old code was trying to match a particular mount point into fstab, namely, the `..../root/.bastille` mountpoint. The original bug is that it left all other custom mount points alone, unrenamed.

So, my take is to sed them all, which makes the rest of the code useless.

The linprocfs/linsysfs special case should also be covered, although I don't understand why the leading dot in the string matched in `sed -i '' "s|.${bastille_jailsdir}/${TARGET}/|${bastille_jailsdir}/${NEWNAME}/|" "${FSTAB_CONFIG}"`.

Hope this helps clarify.

Reply:
Yes I see what you mean. 
However, what happens if a jail is named the same as a directory in the path specified?

Say you have a directory mounted into a jail named test, and the jail is also named test?

What do you think would be the best way to solve this?

Reply:
How would that match `${bastille_jailsdir}/${TARGET}` or `${bastille_jailsdir}/${NEWNAME}`? You'd have to have a path, say, /usr/local/bastille/jails/foo *under* `bastlle_jailsdir`, so your mount point could be:


```
/host/path /usr/local/bastille/jails/TARGET/usr/local/bastille/jails/TARGET nullfs ro 0 0
```

In this case, yes, my approach would rename both TARGET to NEW, while probably you only want to rename the first.

I'd say it's still safer than what we're doing now...

Reply:
Ahhh yes, I see what you mean. Yes that would definitely be best in this particular case.

Reply:
I have no idea.. but I think this file is sourced when you do `bastille rename`

Reply:
It looks like the only place it's called is here. I'm good with this change, unless @cedwards has a good reason to retain the original code.

Reply:
@tschettervictor did you test this code on your system also??

Reply:
Still working on that...

Reply:
My testing confirms that the old code does not rename the mount points at all.
However, I would keep the old code.

In the "clone.sh" function this line
`sed -i '' "s|${bastille_jailsdir}/${TARGET}/root/|${bastille_jailsdir}/${NEWNAME}/root/|" "${FSTAB_CONFIG}"`
is added, which is not present in the "rename.sh"

I would argue that we should just make the "rename" code identical to the "clone" code as they attempt to do the exact same thing.
The current code does in fact change the RELEASE mount point, but does not touch any other custom mount points.

Here is the code from the "clone.sh"

```
 update_fstab() {
    # Update fstab to use the new name
    FSTAB_CONFIG="${bastille_jailsdir}/${NEWNAME}/fstab"
    if [ -f "${FSTAB_CONFIG}" ]; then
        FSTAB_RELEASE=$(grep -owE '([1-9]{2,2})\.[0-9](-RELEASE|-RELEASE-i386|-RC[1-9]|-BETA[1-9]|-CURRENT)|([0-9]{1,2}(-stable-build-[0-9]{1,3}|-stable-LAST))|(current-build)-([0-9]{1,3})|(current-BUILD-LATEST)|([0-9]{1,2}-stable-BUILD-LATEST)' "${FSTAB_CONFIG}" | uniq)
        FSTAB_CURRENT=$(grep -w ".*/releases/.*/jails/${TARGET}/root/.bastille" "${FSTAB_CONFIG}")
        FSTAB_NEWCONF="${bastille_releasesdir}/${FSTAB_RELEASE} ${bastille_jailsdir}/${NEWNAME}/root/.bastille nullfs ro 0 0"
        if [ -n "${FSTAB_CURRENT}" ] && [ -n "${FSTAB_NEWCONF}" ]; then
            # If both variables are set, update as needed
            if ! grep -qw "${bastille_releasesdir}/${FSTAB_RELEASE}.*${bastille_jailsdir}/${NEWNAME}/root/.bastille" "${FSTAB_CONFIG}"; then
                sed -i '' "s|${FSTAB_CURRENT}|${FSTAB_NEWCONF}|" "${FSTAB_CONFIG}"
            fi
        fi
        # Update additional fstab paths with new jail path
        sed -i '' "s|${bastille_jailsdir}/${TARGET}/root/|${bastille_jailsdir}/${NEWNAME}/root/|" "${FSTAB_CONFIG}"
    fi
}
```
I would just add the last line here to the rename function, but keep the old code in place just in case.

Reply:
At the same time, the original code was trying to do something as simple as rename the entire RELEASE fstab, when it is far easier to just rename all occurrences of the old jail to the new jail.

Unless @cedwards was trying to do something specific with it...

Reply:
@tschettervictor @yaazkal @gahr 

so what is the final fix here ??????

Reply:
In my opinion this PR is a valid fix.
It simply renames the jail path of any fstab entry. See below

```
/usr/local/bastille/releases/14.1-RELEASE /usr/local/bastille/jails/jail10/root/.bastille nullfs ro 0 0
/mnt/test /usr/local/bastille/jails/jail2/root/mnt/test nullfs ro 0 0
```
It would only touch the "/usr/local/bastille/jails/**jailname**/root/" part of the fstab. I don't think there is much danger of touching anything else.

It seems the original code tries to validate that the RELEASE actually exists. This might be overkill, because if the original jail is present then it is to be assumed the RELEASE is also present.
I'm good with this PR, it seems to do ok with my tests.

I'd retain the old code just in case...

Reply:
But just to make sure it catches the right code, I went ahead and added an intentional space in front so it will only match the jail path and never the host path.

```
update_fstab() {
    # Update fstab to use the new name
    FSTAB_CONFIG="${bastille_jailsdir}/${NEWNAME}/fstab"
    if [ -f "${FSTAB_CONFIG}" ]; then
        # Update fstab paths with new jail path
        sed -i '' "s| ${bastille_jailsdir}/${TARGET}/root/| ${bastille_jailsdir}/${NEWNAME}/root/|g" "${FSTAB_CONFIG}"
    fi
}
```
I'd recommend this change here as well.

Reply:
Well aaaactually.. if you want the extra check to only match the second field in fstab, you should consider that fields can be separated by spaces or tabs. You version won't rename a tab-separate fstab entry.

Reply:
Good point. Reverted. Thanks.

I tested it and you are correct.

Reply:
I think we can go ahead and merge this. The recent "clone.sh" did exactly the same thing, and it is working.